src/System.Text.Json/tests/Serialization/Value.ReadTests.cs

```diff
@@ -34,6 +34,31 @@ public static void ReadPrimitives()
             Assert.Equal("Hello", s2);
         }
 
+        [Fact]
+        public static void ReadPrimitivesWithWhitespace()
+        {
+            int i = JsonSerializer.Parse<int>(Encoding.UTF8.GetBytes(@" 1 "));
```

Comment:
nit: Just for completeness, can we add tests with bool true/false primitives as well?